ext/pgsql/pgsql.c

```diff
@@ -4820,7 +4819,7 @@ PHP_PGSQL_API zend_result php_pgsql_convert(PGconn *pg_link, const zend_string *
 				}
 				PGSQL_CONV_CHECK_IGNORE();
 				if (err) {
-					php_error_docref(NULL, E_NOTICE, "Expects NULL or IPv4 or IPv6 address string for '%s' (%s)", Z_STRVAL_P(type), ZSTR_VAL(field));
+					zend_value_error("Field \"%s\" must be of type IPv4|IPv6|null, given %s", ZSTR_VAL(field), Z_STRVAL_P(type));
```

Comment:
As this is a value error instead of a type error I think may need to be split into two different cases:
One that handles the types and the second one that checks that a string is a valid IP address.